src/Notes.cc

```diff
@@ -258,14 +257,15 @@ Notes::dump(StoreEntry *entry, const char *key)
         n->dump(entry, key);
 }
 
-const char *
+SBuf
 Notes::toString(const char *sep) const
 {
     static SBuf result;
+    SBuf separator(sep);
```

Comment:
This adds a memory allocation that is not really necessary. I am pretty sure the whole thing needs to be converted to use std::ostream (instead of SBuf) and asList(), but I do not have the time to provide a blueprint right now. I can suggest starting with removing this performance regression.

Reply:
I argue it is not a significant performance regression; the alternative is to call strlen(sep).
In practical terms, this code has only one caller, where the separator is not even specified so the default ("\r\n") is used

Reply:
I wasn't aware of AsList(). It's nice; I'll use it